app/controllers/sign_in/service_account_controller.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module SignIn
+  class ServiceAccountController < SignIn::ApplicationController
```

Comment:
Can you call this `service_account_application_controller`, and inherit from `ActionController::API`

Reply:
And then define the controller with relevant modules with `include` (see how `sign_in/application_controller.rb` does it with `SignIn::Authentication`), basically this means moving the service account authentication logic into a module that the `service_account_application_controller` includes, and then having different module includes for other necessary functionality for the controller

Reply:
This probably largely means just copying the `sign_in/application_controller` with some minor changes but I think that's fine